channeldb/db.go

```diff
@@ -658,19 +659,42 @@ func (c *ChannelStateDB) fetchNodeChannels(chainBucket kvdb.RBucket) (
 func (c *ChannelStateDB) FetchChannel(tx kvdb.RTx, chanPoint wire.OutPoint) (
 	*OpenChannel, error) {
 
+	selector := func(chainBkt walletdb.ReadBucket) ([]byte, *wire.OutPoint,
+		error) {
+
+		var targetChanPoint bytes.Buffer
+		err := writeOutpoint(&targetChanPoint, &chanPoint)
```

Comment:
This should be done outside of the callback to save a couple of bytes in allocation.

Reply:
noice 🔥